docs/src/core_concepts/executor.md

```diff
@@ -40,7 +40,7 @@ Another feature of the `ForkserverExecutor` to mention is the shared memory test
 
 See AFL++'s [_documentation_](https://github.com/AFLplusplus/AFLplusplus/blob/stable/instrumentation/README.persistent_mode.md#5-shared-memory-fuzzing) or the fuzzer example in `forkserver_simple/src/program.c` for reference.
 
-It is very simple, when you call `ForkserverExecutor::new()` with `use_shmem_testcase` true, the `ForkserverExecutor` sets things up and your harness can just fetch the input from `__AFL_FUZZ_TESTCASE_BUF`
+It is very simple, when you build a new `ForkserverExecutor`, you can call `.use_shmem_testcase(true)` on its `::builder()`. Then the `ForkserverExecutor` sets things up and your harness can just fetch the input from `__AFL_FUZZ_TESTCASE_BUF`.
```

Comment:
Wouldn't it be better to have shmem_testcase as default, unless if you specifically set a filename(?)

Reply:
Reasons why I went for `false` to be the default:  

- the [`ForkserverExecutor`](https://aflplus.plus/libafl-book/core_concepts/executor.html#forkserverexecutor) section of the libafl-book  is written in a way that writing to a file is the default: 
"In normal cases, the mutated input is passed between the forkserver and the instrumented binary via .cur_input file."
- the README of `forkserver_simple` reads like it is intended to be a drop-in replacement for AFL(++). Using shared memory test cases always requires additional setup / modifications within a target application and I wouldn't consider it the to be the default.

Reply:
Wait, an AFL++ target will tell the server during the handshake if it supports Shmem input, so , thinking about it, there is probably no reason for this flag at all?

Reply:
Wait, an AFL++ target will tell the server during the handshake if it supports Shmem input, so , thinking about it, there is probably no reason for this flag at all?

Reply:
https://github.com/AFLplusplus/LibAFL/blob/9450954ef2f401e06df68893a238802c449d60cf/libafl/src/executors/forkserver.rs#L660

The buggy line is here, the information is simply not used. Instead, we should set shmem to true here.